src/Symfony/Bundle/WebProfilerBundle/Controller/ProfilerController.php

```diff
@@ -402,4 +414,17 @@ private function renderWithCspNonces(Request $request, string $template, array $
 
         return $response;
     }
+
+    /**
+     * @return array[]
+     */
+    private function findProfiles(string $ip = null, string $url = null, string $limit = null, string $method = null, string $start = null, string $end = null, string $statusCode = null, string $profileType = null): array
+    {
+        $virtualType = 'request' === $profileType ? '' : $profileType;
+
+        return array_filter(
+            $this->profiler->find($ip, $url, $limit, $method, $start, $end, $statusCode),
+            static fn (array $profile): bool => (string) $virtualType === $profile['virtual_type'],
+        );
```

Comment:
Let's make this work:
```suggestion
        return array_filter(
            $this->profiler->find($ip, $url, $limit, $method, $start, $end, $statusCode),
            static fn (array $profile) => $virtualType === $profile['virtual_type'],
        );
```

Reply:
To make this work we need to hard code 'request' in many places in the storage, maybe we could set it as default value in the Profile instead of null?